script.js

```diff
@@ -1,5 +1,202 @@
+  if (currentGameMode == "Enter Player 1 name"){
+    player1 = input
+    currentGameMode = "Enter Player 2 name"
+    console.log(`Player 1: ${player1}`);
+    myOutputValue = `Welcome ${player1}. <br> Player 2, please enter your name click enter. `;
+    console.log(`currentGameMode: ${currentGameMode}`);
+    return myOutputValue
+  }
+  if (currentGameMode == "Enter Player 2 name"){
+    player2 = input
+    currentGameMode = "Player 1 turn"
+    console.log(`Player 2: ${player1}`);
+    myOutputValue = `Welcome ${player2}. <br> Player 1, press enter to start your roll `;
+    console.log(`currentGameMode: ${currentGameMode}`);
+    return myOutputValue
+  }
+  // 
+  var choiceErrorMsg = function(playerNum){
+    if(playerNum == 1){
+      return `Please enter your choice. <br><br> 
+      dice 1 : ${player1diceResults[0]} <br><br>
+      dice 2 : ${player1diceResults[1]} <br><br>
+      Please choose the order of desired dice result. <br>
+      Type 1 and click enter if you wish for dice 1 to be the first number <br>
+      Type 2 and click enter if you wish for dice 2 to be the first number`}
+    else if (playerNum == 2){
+      return `Please enter your choice. <br><br> 
+      dice 1 : ${player2diceResults[0]} <br><br>
+      dice 2 : ${player2diceResults[1]} <br><br>
+      Please choose the order of desired dice result. <br>
+      Type 1 and click enter if you wish for dice 1 to be the first number <br>
+      Type 2 and click enter if you wish for dice 2 to be the first number`
+    }
+  }
+
+  // Start player's turn
+  // Player will output dice result
+  // Player will choose which dice to put first
+  // ^Create shared function to do 2 things to be reused for player 2
+  if(currentGameMode == "Player 1 turn"){
+    myOutputValue = playerTurn(1)
+    currentGameMode = "Player 1 choice"
+    console.log(`currentGameMode: ${currentGameMode}`);
+    return myOutputValue
+  }
+  if(currentGameMode == "Player 1 choice" && input != ""){
+    player1DiceOrder = input
+    myOutputValue = playerTurn(1, player1DiceOrder,player1diceResults)
+    currentGameMode = "Player 2 turn"
+    console.log(`currentGameMode: ${currentGameMode}`);
+    return myOutputValue
+  }
+  else if(currentGameMode == "Player 1 choice" && input == ""){
+    currentGameMode = "Player 1 choice"
+    return choiceErrorMsg(1)
+  }
+  if(currentGameMode == "Player 2 turn"){
+    myOutputValue = playerTurn(2)
+    currentGameMode = "Player 2 choice"
+    console.log(`currentGameMode: ${currentGameMode}`);
+    return myOutputValue
+  }
+  if(currentGameMode == "Player 2 choice" && input != ""){
+    player2DiceOrder = input
+    myOutputValue = playerTurn(2, player2DiceOrder,player2diceResults)
+    currentGameMode = "Game Outcome"
+    console.log(`currentGameMode: ${currentGameMode}`);
+    return myOutputValue
+  }
+  else if(currentGameMode == "Player 2 choice" && input == ""){
+    currentGameMode = "Player 2 choice"
+    return choiceErrorMsg(2)
+  }
```

Comment:
nice abstraction of the player's turn into a separate helper function, makes this more readable! same point again, a bit repetitive, consider how this can be refactored. notice how lines 159-175 are very similar to lines 176-192, except for the player num. 

the reason why i emphasise the point of repetition so much is because in programming, we try not to repeat ourselves if possible - the reason for this is 1. for readability purpose, and 2. for code maintainability purpose, imagine you notice a bug here and need to change something, then you need to change it twice for the code sections you repeated. And this gets more complex if there are more instances of unnecessary repetition in your code.